### PR TITLE
#207: Simple log file appender

### LINKED ISSUE #207: Improve DB error handling

### PR DESCRIPTION
**Pull request by simerplaha**
## Overview

- Resolves #207
- Added simple file appender the archives logs after 10MB.
   image 
- Checked all conditions mentioned in the issue and they are logged as expected.
- Switch file log pattern to log without funky colours (I don't have any log viewer that renders colours well).
- Removed recovery to `RuntimeException` from `DBRunner`. [This](https://github.com/alephium/explorer-backend/commit/7860e1d1bdf63542fe36991e643b494a6369963f) commit adds an `ExplorerError` replacing `RuntimeException` but I don't think we have a need for such a thing right now, so it's removed.

## TODO

Do we need to configure the log directory via some system configuration? The directory config is named `LOG_HOME`.

### CODE REVIEW

File: app/src/main/scala/org/alephium/explorer/persistence/DBRunner.scala [hunk 1]
```diff
@@ -42,19 +37,14 @@ object DBRunner {
...
   /** Temporary function until all things are made stateless */
   @SuppressWarnings(Array("org.wartremover.warts.Overloading"))
   def run[R, E <: Effect](action: DBAction[R, E])(
-      implicit executionContext: ExecutionContext,
-      databaseConfig: DatabaseConfig[PostgresProfile]): Future[R] =
-    databaseConfig.db.run(action).recoverWith {
-      case error => Future.failed(DBRunnerFailed(error))
-    }
+      implicit databaseConfig: DatabaseConfig[PostgresProfile]): Future[R] =
+    databaseConfig.db.run(action)
```
Comment (tdroxler) [inline app/src/main/scala/org/alephium/explorer/persistence/DBRunner.scala:48]: so I guess `db.run` is already failing the future with an Exception?

Comment (simerplaha) [reply app/src/main/scala/org/alephium/explorer/persistence/DBRunner.scala:48]: Yep that's true. [Here](https://github.com/slick/slick/blob/main/slick/src/main/scala/slick/basic/BasicBackend.scala#L76) Slick makes sure that `Exception`s fail the `Future`.

Comment (tdroxler) [reply app/src/main/scala/org/alephium/explorer/persistence/DBRunner.scala:48]: great :+1:

**Review by tdroxler — APPROVED** [review]

File: app/src/main/resources/logback.xml [hunk 2]
```diff
@@ -1,14 +1,41 @@
...
+        <file>${LOG_HOME}/explorer.log</file>
+
+        <filter class="ch.qos.logback.classic.filter.ThresholdFilter">
+            <level>DEBUG</level>
+        </filter>
+
+        <rollingPolicy class="ch.qos.logback.core.rolling.SizeAndTimeBasedRollingPolicy">
+            <fileNamePattern>${LOG_HOME}/archived/explorer.%d{yyyy-MM-dd}.%i.log.gz</fileNamePattern>
+            <!-- max log file size before starting a new log file -->
+            <maxFileSize>10MB</maxFileSize>
```
Comment (polarker) [inline app/src/main/resources/logback.xml:24]: Let's add a totalSizeCap like here: https://github.com/alephium/alephium/blob/master/flow/src/main/resources/logback.xml#L54-L59

Comment (polarker) [reply app/src/main/resources/logback.xml:24]: Maybe we could use the exact same log file as full node, so we could reuse them and improve them in the same way

Comment (tdroxler) [reply app/src/main/resources/logback.xml]: :+1: but I agree with Simer, for log file we should deactivate colors.

Comment (polarker) [reply app/src/main/resources/logback.xml]: Agree. Let's remove for full node too.

**Comment by simerplaha** [comment]
## Log compression

Just a note for documentation purpose. I'm sure this is already known. Copied `logback.xml` doesn't compress archived logs. 

Benefit: A random log like the following (with not many common bytes to compress) reduces from 625.6 MB to 456.3 MB. Actual logs should compress much better. 

Cost: Compression is expensive in terms of time specially for 100mb log files and slf4j is blocking.

Compression is not suited for Explorer.

<details><summary>Sample log</summary>

> 2022-07-29 12:31:51,476 [main]          INFO  o.a.explorer.Simer$                    - LwW84OISeJfwNwA9ofXRovDXZNvLND7ZtVq509tj8mKuLJ3uiw20tqVlmOPT5ZahtprTq8U06WHODqfkJ0RJ8KZOjipWQtXKZz06T4uZCD2wGy6ZgecNPZNKmnaAuzE53hJJdhS62A80PrlSZcsWk2UaXRtZ7NwvcPCr0j6SzZhoAl3sELXUB7HYsNa5G9LW7IXrLzHGBjtQbanrWRzQfvE7iEsThQa27pa4y9SUUxsEOPcvMVrtxCIomX1H8ckp3VndPegZrealPnI7rmiba7FnxGmrOaWOX4kMdBgTJPhzNUCqM7Seqe7QkyrznUxYNlAE6G7ULhpabnBW5jSzjdtALuxPE5NgWLh1Ua6oKpX770uQPOmgrMgAgRtBlHd9Mpn9MI7p0ltzN7q197SMjrmE9VfiG5MrVBXzQ8MRsfjCD0BlzgJNQisdQf7K5WRAhjaweW1o8sJ65KoEGOn9zjB0hXiFu2jcxQuEEF7Ax8np4f4Z2e3LH0oDIawYszuA21iXBWiPkVKiqMYfwzQA9fmWanoRkbNFhj4AbV9AdWcSRhhGrh1WYOsG5n38AwfK2F23Eot9gz9DuiP5Fvb3jJTA91IO8KbOYw1SzzLntxBT0bjhwaYUHnVCr92oknyhab1S2jMzCLzngH2IITPZIkdvAkNNkBDZACw24AdXk6HRUmbk9MI9mNyMVqu0duMWL7kwbngMGJWph1jwFHnMawcLmI77Xf2T4npM6a8SHTfikAJf1duQouPeIX0aZfi1Pa9Ama4jjUfV5JSt9fQu3XOKgGBfvuGx9nWv2fwZLiP8Kv4FhkXOTh1mYnAafrQzFKfqwFXAOzTvtMcPI3MhWaTQjvSA4EZf6Og3IWZiDJMD4b1JZi4ahevZqlBVZDnA4hxuXwDGI3M1MeZBmtIjNp5IK49XdjxfUsVsFH9XjM7HFArhcxHcnXgRLEGFjGyXDC0KuvxKWDdDU36tIdrFgClQh96IePHmdk9ehccn
2022-07-29 12:31:51,476 [main]          ERROR o.a.explorer.Simer$                    - No3CbvFBlWD1NWZOLe8OrQcqthRgkSQevS3uwo7l94JXbLr6tc83lDkE54mEhTGrtnbzwiDnYU5FW1m6zn4MhMVfCm2pIPyPNgta35oecmEHOeXeJL4nPbrtHyioc0Mb8HzylZaY0f85yk25aBI8BWaMHdUb9X7O6dAlGUm0Ozw05WyoNeXcjHPLDes4VsNJqtr7IUF51Vz9zyQZhLf2Z16dtycC9mFBJi2ec4mhTcPi1VwdUDIuH4vMpLQEZj1vCDWRfJ0kTD2n29P7tjcw5jFv4CQRZyKa6NSfhPLB8VRr9NDFHRrNK8Dt2nkm2ztm3v80KB4z89GB2hAmOO5XYYZU9xtKaVGKdYNMazs0I2tZrjlEt57Z7Z9kdmX77jCDthZivwsrNdAXczBTHRB3a2RM1re8sl2rWXbfKuzTytT42Jo1zyVndus7wvJdV1hcppst5wnsbi2orn6ctChwmJfxVP3SgBWdHmVKWstW93SV5TJcL89pUECv0u8vjN2pgbcYk4QwPOTS5dem9G0AF0XQAJPXt2IPXOeS52Nxp6tijjkbIEOJzNyaupMJ8ZCTjW5Ra2vxS30Vxp75c6BTPh2hMpixSGfzwQnbJkczBuuVatnI9qjGdMC2VateqWKwNx2uguf2C3GzyQmGUZU2zUMcIdKcCYH3Zp2pO0jiM6KB5zrJmbaim2y535mL2YbK65GIcqDVGm2NyMvLnywHMpdCdjxgfraNcLG7pN2G0TYAvyEBc1kjN9niX0dXb1ghm4wSa7W2fbqV2NggGDGyzXhLDg7DGlhHInw7I96nS1iyN7yO00rjPYKxuMgitp3dDFqaqTCmdYlffQOjrHuhJEejx3eaocd4xXx5sra53azPECnaPIDnzY6qg2yUpgn3syrEWUE2RjEl22OezrfKgy5sIBMX9PeCTJ0tFiYFXcC4YW92HLeZEkklsZhRA7hEvQwYUbx2aRRFkTXiuP60gsuYJEJC2MnLmGtfJh5P
2022-07-29 12:31:51,476 [main]          WARN  o.a.explorer.Simer$                    - zAix8SrIvMuD4tAo1T3d24PzRfAxfnB4F8hUfJQL4FaAaPFSpHucSPxkz4DM3Y6DuJaYtzhMkdHTQFJccHuymdwQw91uOEZex6LEyEh383MuIfnu8mbFu3ntHXTccLA6HbqrfiLZhA6VbtwPeNCz37ZUFG0pDTMFOlWmY3dm3VydBu7ERVhxHWJUVOBzfy7yLm7457USzL2CvSf6VqEs64Z7IIWarX90n3eSiDrfCNvuXQWx3LD6XFIVeEMU19EBJzc24pW9yD9YhCRv2MSjG4oUbUrqJcFp59ek04WgEWaXygxQSQRu9HNmon9zOVOnRy4P1TxVNeYZG62BypKBtCtZgQX1Ycxgq9v2FqXrfQC2BZHq1yGGIJ0OIStST0hq9XFjpPXLn082XP0BQjPyI9Fxlm2LvVDCtbF84gIb4zX093iCPZ9Mmhyp76LpNcBBpUO1i4OlOpazXCbWvE18Adunw1SwSDfRnYypLyHCKGIHByTluZ56orszTtzgFE4OWaOFYRN2j5cyMbLYJzk90Vq2GfdfkXYx3aVdIkBsMt6pg7tlHvwDBxBefkumvedrwtTXJR0TBLb1JhSAEmIW5qSEQgaekVO8sJE0NkMdhKhCkgc3i1x8mfE7MqdzAGFLgDxBoB5h5TEIJJmthvrZPiOvM2vb8vZn4vDYoFJltjF5GbzU59blKOd6tAOV3QcezyeJp176oKWqYQpwyCSMxeQ1Nbrmd35wAMfg5TSOSi4G9gn3IcYk7UxzrmaFOWTUPEiI70kTOuGRJFIofyd1QtpuOMBtydsntBKK18DNjkQhyWNm7sAZM3X14uSVf79hp0vwCPMDqfKwLdL6zTKIwGoyQhXfSIHDtAIGeFPHaPhhenfIJ7pbYieEdU1zCBjligsfnx1Bj7Vf16nQdV2C2BIWJ6CudzW9Vpr5j7hndHwqT3kFYwodBhs6VyQjJ2LRu1hs6khWbWcJ801gJBPDBqwcqxjvMdhb6Yt6QmYN
</details>

**Pull request merged**